### ISSUE TITLE
ocamldebug segfault on install_printer using module aliasing

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7258]
**Original bug ID:** 7258
**Reporter:** pboutill
**Status:** resolved (set by @xavierleroy on 2017-02-18T17:16:50Z)
**Resolution:** fixed
**Priority:** normal
**Severity:** minor
**Version:** 4.03.0
**Target version:** 4.05.0 +dev/beta1/beta2/beta3/rc1
**Fixed in version:** 4.05.0 +dev/beta1/beta2/beta3/rc1
**Category:** tools (ocaml{lex,yacc,dep,debug,...})
**Monitored by:** @gasche

## Bug description

Consider the file install_printer.ml:
```
module type B =
  sig
    type t
    val print : Format.formatter -> t -> unit
  end

module Foo :
sig
  module Bar : B
end =
  struct
    module Bar : B =
      struct
        type t = int
        let print = Format.pp_print_int
      end
  end

module FooBar = Foo.Bar
```

```
$ ocamlc -c install_printer.ml
$ ocamlc -debug -o a.out install_printer.ml
$ ocamldebug a.out
(ocd) load_printer "install_printer.cmo"
File ./install_printer.cmo loaded
```

Now if you ```install_printer Install_printer.Foo.Bar.print``` everything is fine BUT
if you ```install_printer Install_printer.FooBar.print``` it segfault under MacOS and Debian with 4.02 and 4.03

**Comment by vicuna** [comment]
**Comment author:** @xavierleroy

Tentative fix in commits [trunk b350002] and [4.05 b43beee], will be in release 4.05.

**Comment by vicuna** [comment]
**Comment author:** nbb

i saw i think the same bug with Lacaml printers: 

directory ~/.opam/4.04.0+flambda/lib/lacaml

load_printer bigarray.cma
load_printer lacaml.cma


install_printer Lacaml.Io.pp_fmat segfaults while
install_printer Lacaml__io.pp_fmat works fine.
